main.go

```diff
@@ -737,6 +740,45 @@ func (u *User) changeRoom(r *Room) {
 	u.room.broadcast("", Green.Paint(" --> ")+u.Name+" has joined "+Blue.Paint(u.room.name))
 }
 
+func (u *User) formatPrompt() {
+	u.formattedPrompt = ""
+	last_escaped := false
+	for _, c := range u.Prompt {
+		if c == '\\' {
+			last_escaped = true
+		} else if last_escaped {
+			last_escaped = false
+			switch c {
+			case 'u':
+				u.formattedPrompt += u.Name
+			case 'w':
+				u.formattedPrompt += copyColor(u.room.name, u.Name)
```

Comment:
I'm trying to think about whether a copyColor function is necessary. This could be replaced by a call to applyColorToData(<data>, u.Color, u.ColorBG). We wouldn't need a whole new function with possible new hard-to-track-down ANSI bugs. We've already used applyColorToData in all other such cases (such as when displaying pronouns).

Reply:
As of now, there is no way to know the real color of the username based on the values of u.Color and u.ColorBG. Indeed such color scheme such as a list of color are using a random value when the name is colored. In cases such as with the rainbow color, the random value changes wildly the color rendering.

In such case, the color of the name and hostname can be very different, resulting in quite a ugly render.

Reply:
One way to fix that would be to generate a rng seed before coloring a name and storing that seed in the user info. But I feel like this would be a bit of an overkill for our problem.